src/backend/distributed/utils/metadata_cache.c

```diff
@@ -677,6 +722,32 @@ InvalidateDistRelationCacheCallback(Datum argument, Oid relationId)
 			cacheEntry->isValid = false;
 		}
 	}
+
+	/* now check whether we're in the middle of dropping the extension */
```

Comment:
They're already invalidated, the existing code in `InvalidateDistRelationCacheCallback` takes care of that.